datafusion/datasource-parquet/src/source.rs

```diff
@@ -295,46 +296,16 @@ impl ParquetSource {
         self
     }
 
-    fn with_metrics(&self, metrics: ExecutionPlanMetricsSet) -> Self {
-        let mut conf = self.clone();
-        conf.metrics = metrics;
-        conf
-    }
-
     /// Set predicate information, also sets pruning_predicate and page_pruning_predicate attributes
     pub fn with_predicate(
         &self,
-        file_schema: Arc<Schema>,
+        _file_schema: Arc<Schema>,
         predicate: Arc<dyn PhysicalExpr>,
     ) -> Self {
         let mut conf = self.clone();
 
-        let metrics = ExecutionPlanMetricsSet::new();
-        let predicate_creation_errors =
-            MetricBuilder::new(&metrics).global_counter("num_predicate_creation_errors");
-
-        conf.with_metrics(metrics);
         conf.predicate = Some(Arc::clone(&predicate));
 
-        match PruningPredicate::try_new(Arc::clone(&predicate), Arc::clone(&file_schema))
-        {
-            Ok(pruning_predicate) => {
-                if !pruning_predicate.always_true() {
-                    conf.pruning_predicate = Some(Arc::new(pruning_predicate));
-                }
-            }
-            Err(e) => {
-                debug!("Could not create pruning predicate for: {e}");
-                predicate_creation_errors.add(1);
-            }
-        };
-
-        let page_pruning_predicate = Arc::new(PagePruningAccessPlanFilter::new(
-            &predicate,
-            Arc::clone(&file_schema),
-        ));
-        conf.page_pruning_predicate = Some(page_pruning_predicate);
```

Comment:
We could continue to generate these here and then just not use them later, but that seems worse? If we do remove them maybe we should completely do so (remove the methods, remove the struct fields, make it a breaking change).